### DISCUSSION TITLE
How to use 'STREAMING-AWS4-HMAC-SHA256-PAYLOAD' in PuObject

### DISCUSSION THREAD

**Discussion by T-TRz879** [aws/aws-sdk-java-v2#5763] [Q&A]
I couldn't find the option to set this configuration in the SDK.

**Comment by debora-ito** [comment]
@T-TRz879 The SDK will use `STREAMING-AWS4-HMAC-SHA256-PAYLOAD` depending on some factors in the request that will require payload signing.

Can you give us more context about why you need and how you'll use `STREAMING-AWS4-HMAC-SHA256-PAYLOAD`?

**Reply by T-TRz879** [reply]
We need to compress the data. A regular putobject requires specifying the content length. If we use this behavior, we don't need to specify the content length in advance.

**Reply by asztalosdani** [reply]
Can I force the SDK to use `STREAMING-AWS4-HMAC-SHA256-PAYLOAD`? If not, what are the factors in the request that affect this behaviour?
